tests/src/core/memory.cpp

```diff
@@ -43,6 +43,19 @@ void testMalloc() {
   mem = occa::malloc(bytes, hostPtr, {{"use_host_pointer", false}});
   ASSERT_EQ(mem.ptr<int>()[0], value);
   ASSERT_NEQ(mem.ptr<int>(), hostPtr);
+
+  // test zero byte malloc
+  const occa::udim_t noBytes = 0 * sizeof(int);
+  hostPtr = NULL;
+
+  occa::memory noMem = occa::malloc(noBytes);
+  ASSERT_EQ(noMem.ptr(), (void*) NULL);
```

Comment:
This test is failing on a few platforms. I am a bit leery of using `ASSERT_EQ` for pointers here, since it casts to `size_t`. We should probably have a macro for comparing pointer equality using intptr_t or ptrdiff_t.